ADRs/0003-adopt-libp2p-peer-ids.md

```diff
@@ -0,0 +1,98 @@
+# Adopting Libp2p PeerIDs <!-- omit in toc -->
+
+- Status: in review
+- Deciders: Pocket Network team
+- Date: 2023-04-14
+
+**Table of Contents**
+
+- [Summary](#summary-)
+- [Problem Statement](#problem-statement-)
+- [Background](#background-)
+- [Decision Drivers](#decision-drivers-)
+- [Considered Options](#considered-options-)
+- [Decision Outcome](#decision-outcome-)
+    - [Positive Consequences](#positive-consequences-)
+    - [Negative Consequences](#negative-consequences-)
+- [Pros and Cons of the Options](#pros-and-cons-of-the-options-)
+    - [Adopting Libp2p identity/IDs](#adopting-libp2p-identityids)
+    - [Custom identity scheme](#custom-identity-scheme)
+- [References](#references-)
+
+## Summary 
+
+Facing the need for a unique and verifiable identifier for nodes in the peer-to-peer network, we decided to adopt Libp2p identity/IDs and neglect alternative identity schemes, to achieve a reliable and secure identification system, accepting the additional dependency and integration work, because Libp2p provides a proven solution with robust security features.
+
+## Problem Statement 
+
+At the time of writing, there is no formal specification for identity with respect to pocket network nodes/participants.
+Each node MUST be uniquely identifiable/addressable so that it may be correctly routed to.
+Identities MUST be provable (e.g. via cryptographic signature).
+
+### Context 
+
+- We've recently refactored the P2P module to use libp2p for the transport layer. As a result we're already interoperating with libp2p objects and types.
+- We're simultaneously exploring consolidating and simplifying the concept of node identity throughout the codebase wherever reasonable/feasible.
+
+## Background
+
+[Consolidate and Refactor Node Identity - #348](https://github.com/pokt-network/pocket/issue/348)
+
+## Decision Drivers 
+
+- Unique and verifiable node identifiers
+- Secure and reliable identification system
+- Proven solution in the peer-to-peer ecosystem
+
+## Considered Options 
+
+1. Adopting Libp2p identity/IDs
+2. Custom identity scheme
```

Comment:
Is it worth saying "A custom wrapper" around multiaddr?

Reply:
Multiaddr is orthogonal to peer IDs. It's used represent the network address of a node for the purpose of establishing a network transport connection. A peer ID _can be encapsulated in a multiaddr_ to encode identity information in addition to the network address, for the convenience of the consumer. However, this ADR is coming from a different, more peer-ID-centric, perspective where multiaddr isn't specifically relevant.

Examples of "custom identity scheme"s would be something like what we talked about in the decentralized identity protocol hour. Ranging from something simple, like a self-signed public key (key plus signature) or something more sophisticated, like an X509 certificate chain (self-signed or otherwise). Perhaps these examples could be added using the structure you suggested (https://github.com/pokt-network/pocket-network-protocol/pull/32#discussion_r1199513069)?